docs/Technical-Documentation/tech-memos/login-gov-set-integration/login-gov-set-integration.md

```diff
@@ -0,0 +1,358 @@
+        """Handle account-purged event when a user deletes their Login.gov account."""
+        # Find user by Login.gov UUID
+        user = security_event.user
+
+        # Set login_gov_uuid to null and deactivate user
+        user.login_gov_uuid = None
+        user.account_approval_status = AccountApprovalStatusChoices.DEACTIVATED
+        user.is_active = False
+        user.save()
+
+        logger.info(f"User {user.username} Login.gov account was purged. Prepared for potential recreation.")
+        return True
+
+    @classmethod
+    def _handle_password_reset(cls, security_event):
+        """Handle password-reset event."""
+        user = security_event.user
+        logger.info(f"User {user.username} performed a password reset")
+        return True
+
+    @classmethod
+    def _handle_recovery_activated(cls, security_event):
+        """Handle recovery-activated event when user initiates account recovery."""
+        user = security_event.user
+        logger.info(f"User {user.username} initiated account recovery")
+        return True
+
+    @classmethod
+    def _handle_recovery_information_changed(cls, security_event):
+        """Handle recovery-information-changed event when user modifies their recovery methods."""
+        user = security_event.user
+        logger.info(f"User {user.username} changed their recovery information")
+        return True
+```
+
+### Update Authentication Flow
+
+Update the authentication flow to handle account recreation scenarios. If we find a user with an existing email, a different sub, an existing purge event, and no other events with a timestamp after the purge event, then we can guarantee the user re-created and verified their account with Login.gov. When this occurs we immediately update the user's login_gov_uuid and proceed as normal.
+
+```python
+# tdpservice/users/api/login.py
+def handle_user(self, request, id_token, decoded_token_data):
+    """Handle the incoming user."""
+    # Existing code...
+
+    # Authenticate with `sub` and not username, as user's can change their
+    # corresponding emails externally.
+    user = CustomAuthentication.authenticate(**auth_options)
+
+    if user and user.is_active:
+        # Existing code for active user...
+    elif user and not user.is_active:
+        # Existing code for inactive user...
+    else:
+        User = get_user_model()
+
+        # Check if a user with the same email already exists
+        existing_user = User.objects.filter(email=email).first()
+
+        if existing_user:
+            # Check if last security event was account_purged
+            last_security_event = SecurityEventToken.objects.filter(user=existing_user).order_by('-received_at').first()
+            if last_security_event and last_security_event.event_type == 'account_purged':
+                # Update user login_gov_uuid
+                existing_user.login_gov_uuid = decoded_token_data.get('sub')
+                existing_user.save()
```

Comment:
perhaps we could modify the `save` hook here, or otherwise add an alert when these events happen